packages/firestore/test/unit/local/local_store.test.ts

```diff
@@ -1721,6 +1759,11 @@ function genericLocalStoreTests(
         doc('foo/bar', 1, { sum: 1 }, { hasLocalMutations: true })
       )
       .toContain(doc('foo/bar', 1, { sum: 1 }, { hasLocalMutations: true }))
+      .toHaveQueryDocumentMapping(
+        persistence,
+        /*targetId*/ 4,
+        /*expectedKeys*/ documentKeySet(key('foo/bar'))
+      )
```

Comment:
Please also add a test that verified the new GC behavior.  Something like:

- loadBundle()
- toContain(docA)
- toListen() (whatever the name for that API is s :-P )
- toReturn(docA)
- toUnlisten()
- toContain (docA)

Reply:
I do not think there is a good way to 'listen' in local_store.test, added an integration test instead.

Reply:
FWIW, you can do this with `afterAllocatingQuery`, but the integration test is fine.